client/src/containers/SidenavItem/__tests__/index.spec.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import configureStore from 'redux-mock-store';
+import { create } from 'react-test-renderer';
+ 
+import SidenavItem from './../../SidenavItem';
+jest.mock('react-router-dom', () => ({ Link: 'Link' }));
+
+const mockStore = configureStore();
+
+const mockComponent = (initialState = {}, props) => {
+    const store = mockStore(initialState);
+
+    return (
+        <SidenavItem {...props} store={store} />
+    );
+};
+ 
+describe('<SidenavItem />', () => {
+    it('should render component with no login', () => {
+        const initialState = {
+            users: {}
+        };
+        const props = {
+            item: {
+                type: 'type',
+                link: 'link',
+                icon: 'icon',
+                text: 'text'
+            }
+        };
+        const tree = create(mockComponent(initialState, props)).toJSON();
+        
+        expect(tree).toMatchSnapshot();
+    });
+
+    it('should render component when no authenticated', () => {
+        const initialState = {
+            users: {
```

Comment:
Maybe it is better to move these repetitive things to beforeEach in order to make the code cleaner